enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -2822,6 +2825,442 @@ class AdjointGenerator
+                                    ConstantFP::get(orig_ops[0]->getType(), -1),
+                                    ConstantFP::get(orig_ops[0]->getType(), 1)),
+              op);
+          setDiffe(&I, dif0, Builder2);
+        }
+        return;
+      }
+
+#if LLVM_VERSION_MAJOR < 10
+      case Intrinsic::x86_sse_max_ss:
+      case Intrinsic::x86_sse_max_ps:
+#endif
+      case Intrinsic::maxnum: {
+        auto op0 = gutils->getNewFromOriginal(orig_ops[0]);
+        auto op1 = gutils->getNewFromOriginal(orig_ops[1]);
+        Value *cmp = Builder2.CreateFCmpOLT(op0, op1);
+
+        if (!gutils->isConstantValue(orig_ops[0]) &&
+            !gutils->isConstantValue(orig_ops[1])) {
+          Value *dif = Builder2.CreateSelect(cmp, diffe(orig_ops[0], Builder2),
+                                             diffe(orig_ops[1], Builder2));
+          setDiffe(&I, dif, Builder2);
+        } else if (!gutils->isConstantValue(orig_ops[0])) {
+          Value *dif = Builder2.CreateSelect(
+              cmp, ConstantFP::get(orig_ops[0]->getType(), 0),
+              diffe(orig_ops[0], Builder2));
+          setDiffe(&I, dif, Builder2);
+        } else if (!gutils->isConstantValue(orig_ops[1])) {
+          Value *dif =
+              Builder2.CreateSelect(cmp, diffe(orig_ops[1], Builder2),
+                                    ConstantFP::get(orig_ops[0]->getType(), 0));
+          setDiffe(&I, dif, Builder2);
+        }
+
+        return;
+      }
+
+#if LLVM_VERSION_MAJOR < 10
+      case Intrinsic::x86_sse_min_ss:
+      case Intrinsic::x86_sse_min_ps:
+#endif
+      case Intrinsic::minnum: {
+        auto op0 = gutils->getNewFromOriginal(orig_ops[0]);
+        auto op1 = gutils->getNewFromOriginal(orig_ops[1]);
+        Value *cmp = Builder2.CreateFCmpOLT(op0, op1);
+
+        if (!gutils->isConstantValue(orig_ops[0]) &&
+            !gutils->isConstantValue(orig_ops[1])) {
+          Value *dif = Builder2.CreateSelect(cmp, diffe(orig_ops[0], Builder2),
+                                             diffe(orig_ops[1], Builder2));
+          setDiffe(&I, dif, Builder2);
+        } else if (!gutils->isConstantValue(orig_ops[0])) {
+          Value *dif =
+              Builder2.CreateSelect(cmp, diffe(orig_ops[0], Builder2),
+                                    ConstantFP::get(orig_ops[0]->getType(), 0));
+          setDiffe(&I, dif, Builder2);
+        } else if (!gutils->isConstantValue(orig_ops[1])) {
+          Value *dif = Builder2.CreateSelect(
+              cmp, ConstantFP::get(orig_ops[0]->getType(), 0),
+              diffe(orig_ops[1], Builder2));
+          setDiffe(&I, dif, Builder2);
+        }
+
+        return;
+      }
+
+      case Intrinsic::fma: {
+        Value *dif = nullptr;
+
+        auto op1 = gutils->getNewFromOriginal(orig_ops[1]);
+        auto op2 = gutils->getNewFromOriginal(orig_ops[2]);
+
+        if (!gutils->isConstantValue(orig_ops[0])) {
+          dif = diffe(orig_ops[0], Builder2);
+        }
```

Comment:
Might as well set this to be zero otherwise, and thus simplify the conditional operands below.